### PR TITLE
docs: Adding CocoaPods to iOS plugin

### LINKED ISSUE #10246: Provide more details on adding pods to plugin
### What information needs to be added?

Installing private pods to a plugin. I wrote out my issue in detail on this [Stack Overflow post](https://stackoverflow.com/questions/78111914/adding-private-pod-to-flutter-package)

As a gist, there is never a need to run any `source` or `pod` commands in the Podfile of the projects that implement plugins with Cocoapod dependencies, but for some reason, thats the only way that works for me with my private pod.

What is the correct way of doing it?

### Where should this new content appear?

https://docs.flutter.dev/packages-and-plugins/developing-packages#step-2c-add-ios-platform-code-swifthm

I will keep looking at this over the next few days and if I figure it out, will create a PR with these extra details.

Thank you.

### I would like to fix this problem.

- [X] I will try and fix this problem on docs.flutter.dev.

### PR DESCRIPTION
**Pull request by charlieforward9**
_Description of what this PR is changing or adding, and why:_
This PR adds a small guide to correctly add CocoaPods to a plugin that integrates with iOS. Hopefully saving engineers time when creating plugins, as it took em quite some time to set it up and had to ask a [SO question](https://stackoverflow.com/questions/78111914/adding-private-pod-to-flutter-package/78112482#78112482).

_Issues fixed by this PR (if any):_
Closes #10246

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]
PTAL @sfshaza2

File: src/packages-and-plugins/developing-packages.md [hunk 1]
```diff
@@ -371,15 +372,15 @@ with the following specialized content:
 
 **`android/src/main/java/com/example/hello/HelloPlugin.kt`**
 : The Android platform-specific implementation of the plugin API
-  in Kotlin.
+in Kotlin.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:375]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 2]
```diff
@@ -371,15 +372,15 @@ with the following specialized content:
...
 
 **`ios/Classes/HelloPlugin.m`**
 : The iOS-platform specific implementation of the plugin API
-  in Objective-C.
+in Objective-C.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:379]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 3]
```diff
@@ -371,15 +372,15 @@ with the following specialized content:
...
 
 **`example/`**
 : A Flutter app that depends on the plugin,
-  and illustrates how to use it.
+and illustrates how to use it.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:383]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 4]
```diff
@@ -449,6 +451,36 @@ the path will end with `hello/darwin/Classes` instead.)
 
 You can run the example app by pressing the run (&#9654;) button.
 
+##### Add CocoaPod dependencies
+
+To add `HelloPod` with the version `0.0.1`:
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:456]:
```suggestion
Use the following instructions to add `HelloPod` with the version `0.0.1`:
```

File: src/packages-and-plugins/developing-packages.md [hunk 5]
```diff
@@ -449,6 +451,36 @@ the path will end with `hello/darwin/Classes` instead.)
...
+
+1. Specify dependency at the end of `ios/hello.podspec`:
+
+   ```ruby
+   s.dependency 'HelloPod', '0.0.1'
+   ```
+
+   For private pods, refer to [Private CocoaPods](https://guides.cocoapods.org/making/private-cocoapods.html), ensure repo access:
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:464]:
```suggestion
   For private pods, refer to 
   [Private CocoaPods](https://guides.cocoapods.org/making/private-cocoapods.html),
   to ensure repo access:
```

File: src/packages-and-plugins/developing-packages.md [hunk 6]
```diff
@@ -449,6 +451,36 @@ the path will end with `hello/darwin/Classes` instead.)
...
+       :git => "https://github.com/path/to/HelloPod.git",
+       # Alternatively, for pods hosted locally
+       # :path => "file:///path/to/private/repo",
+       :tag => s.version.to_s
+     }`
+   ```
+
+2. Installing the plugin
+
+   - Add plugin in project’s `pubspec.yaml` dependencies.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:478]:
```suggestion
   - Add the plugin in the project’s `pubspec.yaml` dependencies.
```

File: src/packages-and-plugins/developing-packages.md [hunk 7]
```diff
@@ -449,6 +451,36 @@ the path will end with `hello/darwin/Classes` instead.)
...
+   - Run `flutter pub get`.
+   - In project’s `ios/` directory, run `pod install`.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:480]:
```suggestion
   - In the project’s `ios/` directory, run `pod install`.
```

File: src/packages-and-plugins/developing-packages.md [hunk 8]
```diff
@@ -560,10 +592,10 @@ platform-specific implementation language, as shown above. However,
...
 
 {{site.alert.note}}
-  The examples below only apply to non-web platforms. Web
-  plugin implementations are always written in Dart, and use
-  `pluginClass` and `fileName` for their Dart implementations
-  as shown above.
+The examples below only apply to non-web platforms. Web
+plugin implementations are always written in Dart, and use
+`pluginClass` and `fileName` for their Dart implementations
+as shown above.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:598]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 9]
```diff
@@ -665,24 +697,24 @@ This creates an FFI plugin project in the `hello`
 folder with the following specialized content:
 
 **lib**: The Dart code that defines the API of the plugin,
-  and which calls into the native code using `dart:ffi`.
+and which calls into the native code using `dart:ffi`.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:700]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 10]
```diff
@@ -665,24 +697,24 @@ This creates an FFI plugin project in the `hello`
...
 
 **src**: The native source code, and a `CMakeLists.txt`
-  file for building that source code into a dynamic library.
+file for building that source code into a dynamic library.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:703]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 11]
```diff
@@ -665,24 +697,24 @@ This creates an FFI plugin project in the `hello`
...
 
 **platform folders** (`android`, `ios`, `windows`, etc.): The
-  build files for building and bundling the native code
-  library with the platform application.
+build files for building and bundling the native code
+library with the platform application.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:707]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 12]
```diff
@@ -665,24 +697,24 @@ This creates an FFI plugin project in the `hello`
...
 
 ```yaml
-  plugin:
-    platforms:
-      some_platform:
-        ffiPlugin: true
+plugin:
+  platforms:
+    some_platform:
+      ffiPlugin: true
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:717]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 13]
```diff
@@ -694,35 +726,35 @@ such as when FFI is used for the
...
-    platforms:
-      some_platform:
-        dartPluginClass: SomeClass
-        ffiPlugin: true
+plugin:
+  implements: some_other_plugin
+  platforms:
+    some_platform:
+      dartPluginClass: SomeClass
+      ffiPlugin: true
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:734]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 14]
```diff
@@ -694,35 +726,35 @@ such as when FFI is used for the
...
-  plugin:
-    platforms:
-      some_platform:
-        pluginClass: SomeName
-        ffiPlugin: true
+plugin:
+  platforms:
+    some_platform:
+      pluginClass: SomeName
+      ffiPlugin: true
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:744]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 15]
```diff
@@ -694,35 +726,35 @@ such as when FFI is used for the
...
-  * See the documentation in `linux/CMakeLists.txt`.
-  * See the documentation in `windows/CMakeLists.txt`.
+- For Android: Gradle, which invokes the Android NDK for native builds.
+  - See the documentation in `android/build.gradle`.
+- For iOS and macOS: Xcode, via CocoaPods.
+  - See the documentation in `ios/hello.podspec`.
+  - See the documentation in `macos/hello.podspec`.
+- For Linux and Windows: CMake.
+  - See the documentation in `linux/CMakeLists.txt`.
+  - See the documentation in `windows/CMakeLists.txt`.
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:757]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 16]
```diff
@@ -789,6 +822,7 @@ cd ~/dev/mypackage
 
    set FLUTTER_ROOT=~/dev/flutter     # on Windows
 ```
+
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:825]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 17]
```diff
@@ -799,6 +833,7 @@ cd ~/dev/mypackage
 
    %FLUTTER_ROOT%\bin\cache\dart-sdk\bin\dart doc  # on Windows
 ```
+
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:836]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 18]
```diff
@@ -872,13 +907,13 @@ package_1
...
-  should first consider using when writing your app.
-  To learn more,
-  see the [Flutter Favorites program][].
+Have you noticed that some of the packages and plugins
+on pub.dev are designated as [Flutter Favorites][]?
+These are the packages published by verified developers
+and are identified as the packages and plugins you
+should first consider using when writing your app.
+To learn more,
+see the [Flutter Favorites program][].
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:916]: Remove

File: src/packages-and-plugins/developing-packages.md [hunk 19]
```diff
@@ -891,9 +926,9 @@ quality and usability of your package (and to make it
 more likely to achieve the status of a Flutter Favorite),
 consider including the following items:
 
-* Diverse code usage examples
-* Screenshots, animated gifs, or videos
-* A link to the corresponding code repository
+- Diverse code usage examples
+- Screenshots, animated gifs, or videos
+- A link to the corresponding code repository
```
Comment (sfshaza2) [inline src/packages-and-plugins/developing-packages.md:931]: Remove

**Comment by charlieforward9** [comment]
@sfshaza2 sorry about that, I didnt realize my Prettier plugin was changing all of this. It should all be fixed now! Thank you for the review.

**Review by sfshaza2 — APPROVED** [review]
thanks, @charlieforward9! lgtm

**Pull request merged**